### ISSUE TITLE
vmagent Role no permissions to read "secrets"

### ISSUE THREAD

**Issue by RaSerge** [VictoriaMetrics/operator#653]
operator Chart version: 0.22.0
operator version: v0.34.0
We install operator without clusterwide permissions (WATCH_NAMESPACE: "monitoring")
When i create VmAgent in "monitoring" namespace with
```
serviceScrapeNamespaceSelector: 
    matchLabels:
      name: monitoring
```
vmagent pod goes to CrashLoopBackOff status because of error
`User "system:serviceaccount:monitoring:vmagent-test-vmagent" cannot get resource "secrets" in API group "" in the namespace "monitoring"`

Corresponding serviceaccount, role and rolebinding are created by vm-operator, but role don't have permissions to read "secrets"
```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  labels:
    app.kubernetes.io/component: monitoring
    app.kubernetes.io/instance: test-vmagent
    app.kubernetes.io/name: vmagent
    managed-by: vm-operator
  name: monitoring:vmagent-cluster-access-test-vmagent
  namespace: monitoring
rules:
  - apiGroups:
      - ""
    resources:
      - services
      - endpoints
      - pods
      - endpointslices
    verbs:
      - get
      - list
      - watch
  - apiGroups:
      - networking.k8s.io
      - extensions
    resources:
      - ingresses
    verbs:
      - get
      - list
      - watch
```

**Comment by f41gh7** [comment]
Hello, thanks for reporting, operator incorrectly creates role for vmagent with custom-config reloader, which read configuration directly from kubernetes API.

 Current workaround:
1) assign custom service account for `VMAgent`
2) disable custom-reloader with `--set operator.useCustomConfigReloader=false`


 It would fixed soon at the next patch release.

**Comment by RaSerge** [comment]
>     2. disable custom-reloader with `--set operator.useCustomConfigReloader=false`

@f41gh7 Hello and thank you very much for fast reponse. It's sadly but we cannot use this workaround because of a private network with no internet access With useCustomConfigReloader=false vmagent deployment cannot pull reloader image
```
Failed to pull and unpack image "docker.io/prometheus-operator/prometheus-config-reloader:v0.58.0": failed to resolve reference "docker.io/prometheus-operator/prometheus-config-reloader:v0.58.0": failed to do request: Head "https://registry-1.docker.io/v2/prometheus-operator/prometheus-config-reloader/manifests/v0.58.0": dial tcp 34.205.13.154:443: i/o timeout
```
i tried to set VM_CONTAINERREGISTRY to our internal registry but without success. Registry still docker.io. This is the expected behavior?
Thank you.

**Comment by f41gh7** [comment]
An issue must be fixed at [v0.34.1 release](https://github.com/VictoriaMetrics/operator/releases/tag/v0.34.1)
